### ISSUE TITLE
storage: fix TestIntegration_KMS

### ISSUE THREAD

**Issue by tritone** [googleapis/google-cloud-go#1787]
Failing 100% of the time starting today. Example failure: https://sponge.corp.google.com/target?id=241b6037-5043-4631-a154-6d9951806d14&target=cloud-devrel/client-libraries/go/google-cloud-go/continuous/go111&searchFor=&show=ALL&sortBy=STATUS

```=== RUN   TestIntegration_KMS
--- FAIL: TestIntegration_KMS (0.11s)
    integration_test.go:3167: integration_test.go:2735: close write: googleapi: Error 400: Cloud KMS region 'global' not available for use with GCS region 'US'., invalid
```

Initially I was worried that it was caused by https://github.com/googleapis/google-cloud-go/commit/b12378369d7876ccac2d2283c43d205e2363962b , which modified the storage integration tests, but I can't see any connection.

I also can't reproduce the failure locally with my own project. I suspect something happened to change the default bucket location for the project, so I'll look at fixing that and seeing if that fixes the issue. If it's not a quick fix I will put in a skip for this test.

**Comment by dmilan77** [comment]
I also see the same issue
```
gsutil cp file2.txt gs://$DEVSHELL_PROJECT_ID-kms
Copying file://file2.txt [Content-Type=text/plain]...
BadRequestException: 400 Cloud KMS region 'global' not available for use with GCS region 'US'.
```

**Comment by tritone** [comment]
Hi @dmilan77 , thanks for reporting. I did some research on the integration test failure and determined that, while global-region KMS keyrings worked with us-region buckets during the initial beta launch, this is not considered officially-supported behavior and so it's been phased out for projects that were already using it.

See documentation here: https://cloud.google.com/storage/docs/encryption/customer-managed-keys

> You must create the Cloud KMS key in the same location as the data you intend to encrypt. For example, if your bucket is located in US-EAST1, any Cloud KMS key encrypting objects in that bucket must also be created in US-EAST1.

For the integration test, I switched our test project to use a us-region keychain and that resolved the issue. Let me know if that works for you or if you encounter any further issues.
